ably/websocket_internal_test.go

```diff
@@ -95,32 +95,24 @@ func TestWebsocketDial(t *testing.T) {
 	websocketUrl := fmt.Sprintf("ws%s", strings.TrimPrefix(ts.URL, "http"))
 	testServerURL, _ := url.Parse(websocketUrl)
 
-	// Calculate the expected Ably-Agent header used by tests.
-	expectedAgentHeader := AblySDKIdentifier + " " + GoRuntimeIdentifier + " " + GoOSIdentifier()
-
 	tests := map[string]struct {
-		dialProtocol        string
-		expectedPayloadType byte
-		expectedSubprotocol []string
-		expectedResult      *websocketConn
-		expectedErr         error
+		dialProtocol  string
+		expectedErr   error
+		expectedProto proto
 	}{
 		"Can dial for protocol application/json": {
-			dialProtocol:        "application/json",
-			expectedPayloadType: uint8(1),
-			expectedSubprotocol: []string(nil),
-			expectedErr:         nil,
+			dialProtocol:  "application/json",
+			expectedErr:   nil,
+			expectedProto: 0,
```

Comment:
```suggestion
			expectedProto: jsonProto,
```
0 doesn't say very much, we should probably use the alias `jsonProto` here.